src/app/rundown-view/components/minishelf/mini-shelf.component.ts

```diff
@@ -18,41 +19,51 @@ export class MiniShelfComponent implements OnInit, OnDestroy, OnChanges {
   @Input() public videoClipAction: Tv2VideoClipAction | undefined
 
   private readonly defaultAssetForThumbnail: string = 'assets/sofie-logo.svg'
-  protected mediaDuration: number = 0
+  protected media: Media
   private configurationServiceSubscription: Subscription
   private studioConfiguration: StudioConfiguration | undefined
+  private readonly logger: Logger
+  protected mediaCalculatedDuration: number = 0
 
   constructor(
     private readonly actionService: ActionService,
     private readonly configurationService: ConfigurationService,
-    private readonly mediaStateService: MediaStateService
-  ) {}
+    private readonly mediaStateService: MediaStateService,
+    logger: Logger
+  ) {
+    this.logger = logger.tag('MiniShelfComponent')
+  }
 
   public ngOnInit(): void {
     this.configurationServiceSubscription = this.configurationService.getStudioConfiguration().subscribe((studioConfiguration: StudioConfiguration) => {
       this.studioConfiguration = studioConfiguration
     })
-    void this.updateMediaDataDuration()
+    this.updateMedia()
+      .then(this.calculateMediaDuration)
+      .catch(error => this.logger.error(`Failed to update media, error is ${error} .`))
   }
 
-  private async updateMediaDataDuration(): Promise<void> {
-    if (this.segment.metadata?.miniShelfVideoClipFile !== undefined) {
-      let media: Media | undefined = await this.mediaStateService.getMedia(this.segment.metadata?.miniShelfVideoClipFile)
-      if (!media) {
-        return
-      }
-      if (!this.studioConfiguration) return
-      if (media.duration < this.studioConfiguration.blueprintConfiguration.ServerPostrollDuration) {
-        this.mediaDuration = this.NaN
-      } else {
-        this.mediaDuration = media.duration - this.studioConfiguration.blueprintConfiguration.ServerPostrollDuration
-      }
-    }
+  private async updateMedia(): Promise<void> {
+    if (!this.segment.metadata?.miniShelfVideoClipFile) return
+
+    const media: Media = await this.mediaStateService.getMedia(this.segment.metadata?.miniShelfVideoClipFile)
+    if (!media) return
+    this.media = media
+  }
+
+  private calculateMediaDuration(): void {
+    if (!this.segment.metadata?.miniShelfVideoClipFile) return
+    if (!this.studioConfiguration) return
+    if (this.media.duration < this.studioConfiguration.blueprintConfiguration.ServerPostrollDuration) return
```

Comment:
This is in the end going to come down to personal preference, but you could put these conditionals in one if-statement:
```
if (!this.segment.metadata?.miniShelfVideoClipFile
    || !this.studioConfiguration
    || this.media.duration < this.studioConfiguration.blueprintConfiguration.ServerPostrollDuration) {
  return
}
```
I do agree however, that the above conditional might become "too much" to read. In cases like those an option could be to extract it to its own method:

```
private calcuateMediaDuration(): void {
  if (!this.canMediaBeCalculated()) {
    return
  }
  // Calculate operation here
}

private canMediaBeCalculated(): booelan {
  return this.segment.metadata?.miniShelfVideoClipFile
    && this.studioConfiguration
    && this.media.duration >= this.studioConfiguration.blueprintConfiguration.ServerPostrollDuration
}
```
Again, the above comes down to personal preference :) 

Another note: You could get rid of the check if the media.duration is smaller that the ServerPostrollDuration by using `Math.max()`:
```
this.mediaCaluclateDuration = Math.max(this.media.duration - this.studioConfiguration.blueprintConfiguration.ServerPostrollDuration, 0)
```
The above also comes down to personal preference so take it or leave it :)